src/main/java/org/llorllale/cactoos/matchers/TextMatcher.java

```diff
@@ -28,48 +28,63 @@
 
 import org.cactoos.BiFunc;
 import org.cactoos.Text;
-import org.cactoos.text.FormattedText;
 import org.cactoos.text.UncheckedText;
 import org.hamcrest.Description;
 import org.hamcrest.Matcher;
-import org.hamcrest.TypeSafeDiagnosingMatcher;
+import org.hamcrest.TypeSafeMatcher;
 
 /**
  * Generic {@link Matcher} of {@link Text}.
  *
  * @since 1.0.0
  * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
  */
-public final class TextMatcher extends TypeSafeDiagnosingMatcher<Text> {
+public final class TextMatcher extends TypeSafeMatcher<Text> {
 
     /**
      * The matcher to test.
      */
     private final Matcher<String> matcher;
 
     /**
-     * The description of the matcher's actual text.
+     * Ctor.
+     * @param text The text to match against.
+     * @param func Function that compares actual to expected value.
+     * @param expected The description of the matcher's expected text.
      */
-    private final String actual;
+    public TextMatcher(
+        final Text text,
+        final BiFunc<String, String, Boolean> func,
+        final String expected
+    ) {
+        this(text, func, expected, "Text with value");
+    }
 
     /**
      * Ctor.
      * @param text The text to match against.
      * @param func Function that compares actual to expected value.
-     * @param expected The description of the matcher's expected text.
+     * @param expected The prefix of the matcher's expected text.
+     * @param actual The prefix of the actual text.
+     * @checkstyle ParameterNumberCheck (2 lines)
      */
     public TextMatcher(
```

Comment:
@victornoel Could you add a version tag for this new constructor ?

Reply:
@baudoliver7 in cactoos-matchers we only use `@since 1.0.0` everywhere on classes and not on methods. Once v1 will be released, we will start doing it properly. I think actually we should do the same in cactoos maybe... Personally I find those `@since` tags really useless, see https://github.com/teamed/qulice/issues/1036

Reply:
@victornoel got it :)